### ISSUE TITLE
Docs on how to add relabel_configs config to node_exporter jobs

### ISSUE THREAD

**Issue by eskp** [coreos/prometheus-operator#2140]
There is no documentation and/or example for adding a node label to metrics from node_exporter fx. `node_cpu`.

There are a number of issues referencing different solutions. Some talk about application monitoring metrics (pod metrics), some talking about node level metric relabelling. 

We should document a common practice for doing this for both cases.

From what I could find:
`ServiceMonitor` resource has `metricRelabelings` and `relabelings` relabel_configs per endpoint.
But `ServiceMonitor` does not provide access to k8s node role anyway so can't use it with node metrics, only pod level metrics.

`additionalScrapeConfigs` sounds like it might be the way to go here but we need to provide an example because it seems to be an issue that comes up quite often.

Related: https://github.com/camilb/prometheus-kubernetes/issues/88, https://github.com/coreos/prometheus-operator/issues/2123, https://github.com/coreos/prometheus-operator/issues/1166 and a bunch of others.

**Comment by brancz** [comment]
This is intentional, because the node a target may be associated with, is metadata that does not belong on the target itself, instead you should join that information onto any query where you need it from the `kube_pod_info` metric.

**Comment by zot24** [comment]
### TL;DR

For sure some documentation on how to approach this would help I've been dangling around different issues for few days and I'm still uncertain what's the best way to achieve this.

BTW I share thoughts with you in that last comment @eskp and that's the reason why I didn't wanted to modify any query.

---

I'll try to explain now my use case scenario kind of to help others that came across this issue and to ask the community at the same time, if there is a better way to achieve this:

I have few nodes running outside k8s and wanted to add those metrics into the actual set of dashboards/alerts and rules that are related with the cluster's nodes and provided by default by the community resources that we'll find in [prometheus-operator/kube-prometheus](https://github.com/coreos/prometheus-operator/tree/master/contrib/kube-prometheus) so I don't have to reinvent those.

So far I have added the following under my `endpoints` on my `ServiceMonitor`:

```
relabelings:
  - sourceLabels: [__meta_kubernetes_pod_node_name]
    targetLabel: instance
```

Following the advice in this comment https://github.com/coreos/prometheus-operator/issues/1166#issuecomment-447404996 and I get to work the dashboards and rules that use the label `instance` with expressions like this one found in the [kubernetes-nodes](https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/4fc3fb3ca72059040fab5d94a0ac0933697065ee/dashboards/node.libsonnet#L21) dashboard:

```
"expr": "max(node_load1{cluster=\"$cluster\", job=\"node-exporter\", instance=\"$instance\"})",
```

Next I will try to move the `relabeling` logic into the `additionalScrapeConfigs` as most people recommend that approach instead of adding those into the `ServiceMonitor` but still not sure how to work with the `node` labels to process expressions like used in the [kubernetes-use-method-node](https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/4fc3fb3ca72059040fab5d94a0ac0933697065ee/dashboards/use.libsonnet#L100) dashboard:

```
"expr": "node:node_cpu_utilisation:avg1m{cluster=\"$cluster\", node=\"$node\"}",
```

While writing this comment I realize the existence of things like the `nodeExporterSelector` on the [kubernetes-mixin/mixin.libsonnet](https://github.com/kubernetes-monitoring/kubernetes-mixin), which is the library used to generate all the dashbords, alerts and rules, which make me thinks maybe I could just leave the original labels as they are and use my new labels, e.g. `external-node-exporter` in order to create new set of dashboards, alerts and rules for these new resources with minimal effort?

```
local kubernetes = import "kubernetes-mixin/mixin.libsonnet";

kubernetes {
  _config+:: {
    nodeExporterSelector: 'job="kubernetes-node-exporter"',
...
```

As @eskp have mentioned will be great to hear other people opinion.

**Comment by brancz** [comment]
Queries such as

```
node:node_cpu_utilisation:avg1m{cluster=\"$cluster\", node=\"$node\"}
```

Work because the recording rule for the `node:node_cpu_utilisation:avg1m` series, joins that label from the respective kube-state-metrics metrics onto the node-exporter metrics.

I think the right thing here to do is to fold all the metrics available in the "Nodes" dashboard into the USE style dashboards in the kubernetes-mixin, which already has all the "infrastructure" and correct methodology for selecting by nodes.

**Comment by brancz** [comment]
Closing due to inactivity.
